Build failure: linuxKernel.packages.linux_6_14.rtl8852bu

### Nixpkgs version

- Stable (25.05)

### Steps to reproduce

NixOS 

1. having the package in boot.extraModulePackages
2. switch channel from 24.11 to 25.05
3. upgrade
4. build fails on 25.05 channel

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/job/nixos/release-25.05/nixpkgs.linuxKernel.packages.linux_6_14.rtl8852bu.x86_64-linux

### Relevant log output

```console
this derivation will be built:
  /nix/store/zzn2928zf8c3igypbnb2lfnwc6ndzj48-rtl8852bu-6.14.8-unstable-2024-05-25.drv
building '/nix/store/zzn2928zf8c3igypbnb2lfnwc6ndzj48-rtl8852bu-6.14.8-unstable-2024-05-25.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/xh0pn46jrqr8cdn3785balla7wpqlm8f-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
build flags: -j16 SHELL=/nix/store/xy4jjgw87sbgwylm5kn047d9gkbhsr9x-bash-5.2p37/bin/bash ARCH=x86_64 CONFIG_PLATFORM_I386_PC=y CONFIG_PLATFORM_ARM_RPI=n
make ARCH=x86_64 CROSS_COMPILE= -C /nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/build M=/build/source  modules
make[1]: Entering directory '/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/build'
make[2]: Entering directory '/build/source'
  CC [M]  platform/platform_ops.o
  CC [M]  os_dep/osdep_service.o
  CC [M]  os_dep/osdep_service_linux.o
  CC [M]  os_dep/linux/rtw_cfg.o
  CC [M]  os_dep/linux/os_intfs.o
  CC [M]  os_dep/linux/ioctl_linux.o
  CC [M]  os_dep/linux/xmit_linux.o
  CC [M]  os_dep/linux/mlme_linux.o
  CC [M]  os_dep/linux/recv_linux.o
  CC [M]  os_dep/linux/ioctl_cfg80211.o
  CC [M]  os_dep/linux/rtw_cfgvendor.o
platform/platform_ops.c:15:10: fatal error: drv_types.h: No such file or directory
   15 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
os_dep/osdep_service.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
  CC [M]  os_dep/linux/os_ch_utils.o
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: platform/platform_ops.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/osdep_service.o] Error 1
os_dep/osdep_service_linux.c:16:10: fatal error: drv_types.h: No such file or directory
   16 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/osdep_service_linux.o] Error 1
os_dep/linux/ioctl_linux.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
os_dep/linux/rtw_cfg.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/ioctl_linux.o] Error 1
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/rtw_cfg.o] Error 1
os_dep/linux/os_intfs.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
os_dep/linux/xmit_linux.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/os_intfs.o] Error 1
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/xmit_linux.o] Error 1
os_dep/linux/recv_linux.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
os_dep/linux/mlme_linux.c:19:10: fatal error: drv_types.h: No such file or directory
   19 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/recv_linux.o] Error 1
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/mlme_linux.o] Error 1
os_dep/linux/ioctl_cfg80211.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/ioctl_cfg80211.o] Error 1
os_dep/linux/rtw_cfgvendor.c:16:10: fatal error: drv_types.h: No such file or directory
   16 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
os_dep/linux/os_ch_utils.c:17:10: fatal error: drv_types.h: No such file or directory
   17 | #include <drv_types.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/rtw_cfgvendor.o] Error 1
make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/os_ch_utils.o] Error 1
make[3]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/Makefile:1996: .] Error 2
make[2]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/Makefile:251: __sub-make] Error 2
make[2]: Leaving directory '/build/source'
make[1]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/Makefile:251: __sub-make] Error 2
make[1]: Leaving directory '/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/build'
make: *** [Makefile:746: modules] Error 2
error: builder for '/nix/store/zzn2928zf8c3igypbnb2lfnwc6ndzj48-rtl8852bu-6.14.8-unstable-2024-05-25.drv' failed with exit code 2;
       last 25 log lines:
       >       |          ^~~~~~~~~~~~~
       > compilation terminated.
       > make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/recv_linux.o] Error 1
       > make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/mlme_linux.o] Error 1
       > os_dep/linux/ioctl_cfg80211.c:17:10: fatal error: drv_types.h: No such file or directory
       >    17 | #include <drv_types.h>
       >       |          ^~~~~~~~~~~~~
       > compilation terminated.
       > make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/ioctl_cfg80211.o] Error 1
       > os_dep/linux/rtw_cfgvendor.c:16:10: fatal error: drv_types.h: No such file or directory
       >    16 | #include <drv_types.h>
       >       |          ^~~~~~~~~~~~~
       > compilation terminated.
       > os_dep/linux/os_ch_utils.c:17:10: fatal error: drv_types.h: No such file or directory
       >    17 | #include <drv_types.h>
       >       |          ^~~~~~~~~~~~~
       > compilation terminated.
       > make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/rtw_cfgvendor.o] Error 1
       > make[4]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/scripts/Makefile.build:207: os_dep/linux/os_ch_utils.o] Error 1
       > make[3]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/Makefile:1996: .] Error 2
       > make[2]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/Makefile:251: __sub-make] Error 2
       > make[2]: Leaving directory '/build/source'
       > make[1]: *** [/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/source/Makefile:251: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/gc6183aj47phigiznqivk8n0dfa37sdr-linux-6.14.8-dev/lib/modules/6.14.8/build'
       > make: *** [Makefile:746: modules] Error 2
       For full logs, run 'nix log /nix/store/zzn2928zf8c3igypbnb2lfnwc6ndzj48-rtl8852bu-6.14.8-unstable-2024-05-25.drv'.
```

### Additional context

I tried to upgrade from Vicuna (24.11) to Warbler (25.05). It also fails to build on aarch64.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.9, NixOS, 24.11 (Vicuna), 24.11.712909.1dab772dd4a6`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers

@lonyelon 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/rtl8852bu/default.nix
+++ b/pkgs/os-specific/linux/rtl8852bu/default.nix
@@ -9,13 +9,13 @@
 
 stdenv.mkDerivation rec {
   pname = "rtl8852bu";
-  version = "${kernel.version}-unstable-2024-05-25";
+  version = "${kernel.version}-unstable-2025-05-18";
 
   src = fetchFromGitHub {
     owner = "morrownr";
-    repo = pname;
-    rev = "1acc7aa085bffec21a91fdc9e293378e06bf25e7";
-    hash = "sha256-22vzAdzzM5YnfU8kRWSK3HXxw6BA4FOWXLdWEb7T5IE=";
+    repo = "rtl8852bu-20240418";
+    rev = "42de963695ffc7929a4905aed5c5d7da7c1c2715";
+    hash = "sha256-BvOw9MU4eibeMJEOkifKFatCUNGdujNUZav+4D9bYKY=";
   };
 
   nativeBuildInputs = [
@@ -30,7 +30,10 @@ stdenv.mkDerivation rec {
   postPatch = ''
     substituteInPlace ./Makefile \
       --replace-fail /sbin/depmod \# \
-      --replace-fail '$(MODDESTDIR)' "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/"
+      --replace-fail '$(MODDESTDIR)' "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/" \
+      --replace-fail 'cp -f $(MODULE_NAME).conf /etc/modprobe.d' \
+      'mkdir -p $out/etc/modprobe.d && cp -f $(MODULE_NAME).conf $out/etc/modprobe.d' \
+      --replace-fail "sh edit-options.sh" ""
     substituteInPlace ./platform/i386_pc.mk \
       --replace-fail /lib/modules "${kernel.dev}/lib/modules"
   '';
@@ -54,13 +57,19 @@ stdenv.mkDerivation rec {
     nuke-refs $out/lib/modules/*/kernel/net/wireless/*.ko
   '';
 
+  env.NIX_CFLAGS_COMPILE = "-Wno-designated-init"; # Similar to 79c1cf6
+
   enableParallelBuilding = true;
 
   meta = with lib; {
     description = "Driver for Realtek rtl8852bu and rtl8832bu chipsets, provides the 8852bu mod";
-    homepage = "https://github.com/morrownr/rtl8852bu";
+    homepage = "https://github.com/morrownr/rtl8852bu-20240418";
     license = licenses.gpl2Only;
-    platforms = platforms.linux;
-    maintainers = with maintainers; [ lonyelon ];
+    platforms = [ "x86_64-linux" ];
+    broken = kernel.kernelOlder "6" && kernel.isHardened; # Similar to 79c1cf6
+    maintainers = with maintainers; [
+      lonyelon
+      thtrf
+    ];
   };
 }

--- a/pkgs/os-specific/linux/rtl8852bu/default.nix
+++ b/pkgs/os-specific/linux/rtl8852bu/default.nix
@@ -7,7 +7,7 @@
   nukeReferences,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "rtl8852bu";
   version = "${kernel.version}-unstable-2025-05-18";
 
@@ -61,15 +61,15 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  meta = with lib; {
+  meta = {
     description = "Driver for Realtek rtl8852bu and rtl8832bu chipsets, provides the 8852bu mod";
     homepage = "https://github.com/morrownr/rtl8852bu-20240418";
-    license = licenses.gpl2Only;
+    license = lib.licenses.gpl2Only;
     platforms = [ "x86_64-linux" ];
     broken = kernel.kernelOlder "6" && kernel.isHardened; # Similar to 79c1cf6
-    maintainers = with maintainers; [
+    maintainers = with lib.maintainers; [
       lonyelon
       thtrf
     ];
   };
-}
+})

